LukasWaveformReaderApp/src/WaveformReader.cpp

```diff
@@ -35,74 +34,80 @@ WaveformReader::WaveformReader(const char *portName, int bayNumber, int bufferSi
   //We register some useful hardware interfaces our port driver could want to know
   Path p;
   p = cpswGetRoot();
-  
-  _TriggerHwAutoRearm = IScalVal::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/TriggerHwAutoRearm").c_str()));
-  _DataBufferSize = IScalVal::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/DataBufferSize").c_str()));
-  _TrigCount = IScalVal_RO::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/TrigCount").c_str()));
-  _WebInit = ICommand::create(p->findByName(("/mmio/AmcCarrierCore/AmcCarrierBsa/BsaWaveformEngine[" + std::to_string(bayNumber) + "]/WaveformEngineBuffers/Initialize").c_str()));
-  _ClkFrequency = IScalVal_RO::create(p->findByName(("/mmio/AppTop/AppCore/AmcGenericAdcDacCore[" + std::to_string(bayNumber) + "]/AmcGenericAdcDacCtrl/AmcClkFreq").c_str()));
 
+  for (int bayNumber = 0; bayNumber < 2; bayNumber++)
+  {
+    _TriggerHwAutoRearm = IScalVal::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/TriggerHwAutoRearm").c_str()));
+    _DataBufferSize = IScalVal::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/DataBufferSize").c_str()));
+    _TrigCount = IScalVal_RO::create(p->findByName(("/mmio/AppTop/DaqMuxV2[" + std::to_string(bayNumber) + "]/TrigCount").c_str()));
+    _WebInit = ICommand::create(p->findByName(("/mmio/AmcCarrierCore/AmcCarrierBsa/BsaWaveformEngine[" + std::to_string(bayNumber) + "]/WaveformEngineBuffers/Initialize").c_str()));
+    _ClkFrequency = IScalVal_RO::create(p->findByName(("/mmio/AppTop/AppCore/AmcGenericAdcDacCore[" + std::to_string(bayNumber) + "]/AmcGenericAdcDacCtrl/AmcClkFreq").c_str()));
+  }
+  
+  
   //Connecting to the records our port driver will eventually need to interact with
   for(int pvID = 0; pvID < waveformPVs; pvID++)
   {
     //For loop generates the string identifier for each Waveform records and then creates a parameter our asynDriver can interact with for it
     int waveform_param_index;
     int complete_waveform_param_index;
-    int complete_x_axis_waveform_param_index;
+    int complete_z_axis_waveform_param_index;
     int extracted_waveform_param_index;
-    int extracted_x_axis_waveform_param_index;
-    std::string pvIdentifier = "WAVEFORM:" + std::to_string(pvID);
-    std::string complete_pvIdentifier = "COMPLETE_WAVEFORM:" + std::to_string(pvID);
-    std::string complete_x_axis_pvIdentifier = "COMPLETE_X_AXIS:" + std::to_string(pvID);
-    std::string extracted_pvIdentifier = "EXTRACTED_WAVEFORM:" + std::to_string(pvID);
-    std::string extracted_x_axis_pvIdentifier = "EXTRACTED_X_AXIS:" + std::to_string(pvID);
-    std::cout << pvIdentifier << std::endl;
+    int extracted_z_axis_waveform_param_index;
```

Comment:
We shold prob have a general write up in the readme about the z axis, or at least some documentation in the code itself about what we're setting up here